### PR TITLE
DAOS-9008 packaging: Fix rpmlint warnings

### PR DESCRIPTION
**Pull request by brianjmurrell**
Including an incorrect use of "%else if ..." that was an overzealous
attempt at per-OS guarding where we just need to handle a single
outlier.

### CODE REVIEW

**Review by daltonbohning — DISMISSED** [review]
LGTM. Just some questions for my own understanding

**Review by JohnMalmberg — APPROVED** [review]

File: utils/rpms/daos.spec [hunk 1]
```diff
@@ -9,12 +9,12 @@
 %if (0%{?rhel} >= 8)
 # https://bugzilla.redhat.com/show_bug.cgi?id=1955184
 %define _use_internal_dependency_generator 0
-%define __find_requires %{_sourcedir}/bz-1955184_find-requires
+%define __find_requires %{SOURCE1}
```
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply utils/rpms/daos.spec:12]: Yes, `${SOURCE$n}` are the variables that resolve to the respective `Source$n:`s.

File: utils/rpms/daos.spec [hunk 2]
```diff
@@ -452,8 +452,8 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 %{_mandir}/man8/dmg.8*
 
 %files client-tests
-%dir %{_prefix}/lib/daos
-%{_prefix}/lib/daos/TESTING
+%dir %{daoshome}
```
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply utils/rpms/daos.spec:455]: Usually nothing.  For reference:
https://www.gnu.org/prep/standards/html_node/Directory-Variables.html#:~:text=Generally%2C%20%24(exec_prefix)%20is%20used,used%20directly%20for%20other%20directories
https://www.gnu.org/software/automake/manual/html_node/Standard-Directory-Variables.html

[the remark this answers is not in the document]
Comment (daltonbohning) [reply utils/rpms/daos.spec:455]: Thanks!

**Comment by brianjmurrell** [comment]
The test failure looks unrelated: [DAOS-9038](https://daosio.atlassian.net/browse/DAOS-9038)

**Comment by johannlombardi** [comment]
Please merge master.

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**